--- src/adapter_gzip.cc		Wed Jun  8 21:21:10 2016
+++ src/adapter_gzip.cc		Sat Jun 18 22:32:09 2016
@@ -548,7 +548,7 @@
 	
 
 	Must(receivingVb == opOn);
-	receivingVb = opComplete;
+	stopVb();
 	if (sendingAb == opOn) {
 		hostx->noteAbContentDone(atEnd);
 		sendingAb = opComplete;
@@ -611,7 +611,7 @@
 // if the host does not know that already
 void Adapter::Xaction::stopVb() {
 	if (receivingVb == opOn) {
-		hostx->vbStopMaking();
+		hostx->vbStopMaking(); // we will not call vbContent() any more
 		receivingVb = opComplete;
 	} else {
 		// we already got the entire body or refused it earlier
